modules/login/src/main/resources/META-INF/resources/login/js/registration.js

```diff
@@ -0,0 +1,401 @@
+            $('.selectpicker', $formStep1).selectpicker('render');
+            $('#btnNext1').prop('disabled', 'disabled');
+        },0); //Set Timeout so that this code block doesn't get exec'd before the form reset happens.
+
+
+    }
+    function onResetForm2() {
+    }
+    function onResetForm3() {
+        setTimeout(function() {
+            $('.selectpicker', $formStep3).selectpicker('render');
+        },0); //Set Timeout so that this code block doesn't get exec'd before the form reset happens.
+    }
+    function onResetForm4() {
+        setTimeout(function() {
+            $($formStep4).validator('update');
+        },0); //Set Timeout so that this code block doesn't get exec'd before the form reset happens.
+    }
+
+    function populateOrgData() {
+        var orgtypesUrl = endpointRoot + orgtypesService;
+        var orgsUrl = endpointRoot + orgsService;
+
+        $.when($.get(orgtypesUrl),$.get(orgsUrl))
+            .then(function(orgtypesResult, orgsResult) {
+
+                orgLookupData.organizationTypes = createLookup(orgtypesResult[0].organizationTypes); //stuff in global for events to use.
+                orgLookupData.organizations = createLookup(orgsResult[0].organizations);
+
+                var sortableOrgTypes = [];
+                //Push the object into an array for sorting.
+                for (var orgType in orgLookupData.organizationTypes) {
+                    sortableOrgTypes.push([orgType, orgLookupData.organizationTypes[orgType]]);
+                }
+
+                //sort by name before populating the select options.
+                sortableOrgTypes.sort(sortobjArrayByName);
+                //populate the options now that they are sorted properly.
+                populateSelect($($selectOrgType),sortableOrgTypes);
+                $($selectOrgType).selectpicker('refresh'); //Refresh because we just added a bunch of items
+
+                //Extract the organizations of four specific OrgTypes: 'USAR', 'Federal IMT', 'CDF IMT', and 'Other Local IMT'
+                // Put those organizations as options, each having its own dropdown list.
+                populateSelect($($selectImtUSAR),getOrgsData(orgLookupData,"USAR").sort(sortobjArrayByName));
+                $($selectImtUSAR).selectpicker('refresh');
+                populateSelect($($selectImtFed),getOrgsData(orgLookupData,"Federal IMT").sort(sortobjArrayByName));
+                $($selectImtFed).selectpicker('refresh');
+                populateSelect($($selectImtCDF),getOrgsData(orgLookupData,"CDF IMT").sort(sortobjArrayByName));
+                $($selectImtCDF).selectpicker('refresh');
+                populateSelect($($selectImtLocal),getOrgsData(orgLookupData,"Other Local IMT").sort(sortobjArrayByName));
+                $($selectImtLocal).selectpicker('refresh');
+                return true;
+            })
+            .fail(function(result, errortext, error) {
+                console.log(errortext);
```

Comment:
I left this in for production support. True that the customer can't do anything with it, but someone troubleshooting why the end-point call may be failing would find it useful since the browser really doesn't have an offline logging facility.